app/controllers/UploadFormController.scala

```diff
@@ -48,7 +48,7 @@ class UploadFormController @Inject()(
       val uploadId           = UploadId.generate
       val successRedirectUrl = appConfig.upscanRedirectBase +  routes.UploadFormController.showResult(uploadId).url
       val errorRedirectUrl   = appConfig.upscanRedirectBase + "/disclose-cross-border-arrangements/error"
-      val callbackUrl = appConfig.callbackEndpointTarget
+      val callbackUrl = controllers.routes.UploadCallbackController.callback().absoluteURL(appConfig.upscanProtocol == true)
```

Comment:
you're comparing unrelated types: either: upscanProtocol == "https" or preferable create another method in appConfig like upscanUsingHttps which does that comparison